internal/git/gitsync_processor.go

```diff
@@ -76,56 +78,107 @@ func watchRepo(ctx context.Context, restConfig *rest.Config, repo *v1.Repository
-	commit, err := r.CommitObject(*h)
-	if err != nil {
-		logger.Errorw("error checkout the commit", "hash", h.String(), "err", err)
-		return err
+	namespacedName := types.NamespacedName{
+		Namespace: gitSync.Namespace,
+		Name:      gitSync.Name,
 	}
-
-	// Retrieve the tree from the commit.
-	tree, err := commit.Tree()
-	if err != nil {
-		logger.Errorw("error get the commit tree", "err", err)
-		return err
+	gitSync = &v1.GitSync{}
+	if err = k8Client.Get(ctx, namespacedName, gitSync); err != nil {
+		// if we aren't able to do a Get, then either it's been deleted in the past, or something else went wrong
+		if apierrors.IsNotFound(err) {
+			return nil
+		} else {
+			logger.Errorw("Unable to get GitSync", "err", err)
+			return err
+		}
 	}
+	val, ok := gitSync.Status.CommitStatus[repo.Name]
+	// Only create the resources for the first time if not created yet.
+	// Otherwise, monitoring with intervals.
+	if !ok {
+		// Retrieving the commit object matching the hash.
+		commit, err := r.CommitObject(*hash)
+		if err != nil {
+			logger.Errorw("error checkout the commit", "hash", hash.String(), "err", err)
+			return err
+		}
 
-	if !isRootDir(repo.Path) {
-		// Locate the tree with the given path.
-		tree, err = tree.Tree(repo.Path)
+		// Retrieve the tree from the commit.
+		tree, err := commit.Tree()
 		if err != nil {
-			logger.Errorw("error locate the path", "err", err)
+			logger.Errorw("error get the commit tree", "err", err)
 			return err
 		}
-	}
 
-	// Read all the files under the path and apply each one respectively.
-	err = tree.Files().ForEach(func(f *object.File) error {
-		logger.Debugw("read file", "file_name", f.Name)
-		//TODO: this currently assumes that one file contains just one manifest - modify for multiple
-		manifest, err := f.Contents()
+		if !isRootDir(repo.Path) {
+			// Locate the tree with the given path.
+			tree, err = tree.Tree(repo.Path)
+			if err != nil {
+				logger.Errorw("error locate the path", "repository path", repo.Path, "err", err)
+				return err
+			}
+		}
+
+		// Read all the files under the path and apply each one respectively.
+		err = tree.Files().ForEach(func(f *object.File) error {
+			logger.Debugw("read file", "file_name", f.Name)
+			// TODO: this currently assumes that one file contains just one manifest - modify for multiple
+			manifest, err := f.Contents()
+			if err != nil {
+				logger.Errorw("cannot get file content", "filename", f.Name, "err", err)
+				return err
+			}
+
+			// Apply manifest in cluster
+			return client.ApplyResource([]byte(manifest), namespace)
+		})
 		if err != nil {
-			logger.Errorw("cannot get file content", "filename", f.Name, "err", err)
 			return err
 		}
 
-		// Apply manifest in cluster
-		return k8sClient.ApplyResource([]byte(manifest), namespace)
-	})
-	if err != nil {
-		return err
+		commitStatus := v1.CommitStatus{
+			Hash:     hash.String(),
+			Synced:   true,
+			SyncTime: metav1.NewTime(time.Now()),
+		}
```

Comment:
it looks like we won't log a CommitStatus with Synced=false if we fail?

Reply:
yeah, good question, I feel that is not necessary, as no record can mean we   are not synced and we may get errors when we commit status with `synced=false` . But it is a nice to have though. HDYT?

Reply:
It's true that if we have an error involving Kubernetes, then logging commit status probably won't work anyway. But if we have an error involving Git, then it would. Actually, there is an [Error](https://github.com/numaproj-labs/numaplane/blob/main/api/v1/gitsync_types.go#L110) field in the CommitStatus. We should be using that.

Reply:
Ah that's a good point, that's the intention.

Reply:
Okay, please create an Issue